src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -109,6 +99,29 @@ internal partial class Binder
                 return CreateTupleLiteralConversion(syntax, (BoundTupleLiteral)source, conversion, isCast: isCast, conversionGroupOpt, destination, diagnostics);
             }
 
+            if (conversion.Kind == ConversionKind.SwitchExpression)
+            {
+                return ConvertSwitchExpression((BoundUnconvertedSwitchExpression)source, destination, diagnostics);
+            }
+
+            if (source.Kind == BoundKind.UnconvertedSwitchExpression)
+            {
+                TypeSymbol type = source.Type;
+                bool hasErrors = false;
+                if (type is null)
+                {
+                    Debug.Assert(!conversion.Exists);
+                    type = CreateErrorType();
+                    hasErrors = true;
+                }
+
+                source = ConvertSwitchExpression((BoundUnconvertedSwitchExpression)source, type, diagnostics, hasErrors);
+                if (destination.Equals(type, TypeCompareKind.ConsiderEverything) && wasCompilerGenerated)
```

Comment:
>ConsiderEverything [](start = 61, length = 18)

Not sure about the ConsiderEverything. For example, do differences of `dynamic` vs. `object`, or tuple names matter?

Reply:
This method never produced identity conversion before.  See line 74.  I have already opened an issue about that.

---
In reply to: [301826665](https://github.com/dotnet/roslyn/pull/37052#discussion_r301826665) [](ancestors = 301826665)